lib/jsonapi/plugs/underscore_parameters.ex

```diff
@@ -104,4 +104,19 @@ defmodule JSONAPI.UnderscoreParameters do
       conn
     end
   end
+
+  defp replace_query_params(query_params) do
+    # Underscore the keys of all of the query parameters
+    underscored_query_params = JString.expand_fields(query_params, &JString.underscore/1)
+
+    # If the fields[...] query parameter is present, only underscore its values, but not its keys
+    case Map.fetch(query_params, "fields") do
+      {:ok, fields} when is_map(fields) ->
+        fields = Map.new(fields, fn {k, v} -> {k, JString.underscore(v)} end)
+        Map.put(underscored_query_params, "fields", fields)
```

Comment:
Annotation: This is the main change.  If `fields` are present in the original `query_params`, then underscore all of the values, and put it with the rest of the `underscored_query_params`.